sdk/python/kfp/containers/_component_builder.py

```diff
@@ -248,43 +308,132 @@ def build_python_component(
-
-  if python_version == 'python2':
-    import warnings
-    warnings.warn('Python2 is not longer supported')
+  if is_v2:
+
+    program_code = _purge_program_launching_code(
+        commands=command_line_args,
+        python_version=python_version,
+        is_v2=True)
+
+    # Override user program args for new-styled component.
+    # TODO: The actual program args will be changed after we support v2
+    # component on KFP.
+    program_args = []
+    for component_input in component_spec.inputs or []:
+      if component_input._passing_style == components.InputArtifact:
+        # For each input artifact, there'll be possibly 3 arguments passed to
+        # the user program:
+        # 1. {name of the artifact}_input_path: The actual path, or uri, of the
+        #    input artifact.
+        # 2. {name of the artifact}_input_pod_name: The pod ID of the producer.
+        # 3. {name of the artifact}_input_output_name: The output name of the
+        #    artifact, by which the artifact can be found in the producer
+        #    metadata JSON file.
+        program_args.append('--{}{}'.format(
+            component_input.name,
+            entrypoint.INPUT_URI_SUFFIX
+        ))
+        program_args.append(
+            _structures.InputUriPlaceholder(
+                input_name=component_input.name))
+        program_args.append('--{}{}'.format(
+            component_input.name,
+            entrypoint.PRODUCER_POD_ID_SUFFIX
+        ))
+        program_args.append(
+            '{{{{inputs.parameters.{input}}}}}'.format(
+                input=_components.PRODUCER_POD_NAME_PARAMETER.format(
+                    component_input.name)))
+        # TODO(numerology): Consider removing the need of output name
+        # placeholder by letting v2 component output two metadata files per
+        # output.
+        program_args.append('--{}{}'.format(
+            component_input.name,
+            entrypoint.OUTPUT_NAME_SUFFIX
+        ))
+        program_args.append(_structures.InputOutputPortNamePlaceholder(
+            input_name=component_input.name))
+
+      elif component_input._passing_style is None:
+        program_args.append('--{}{}'.format(
+            component_input.name,
+            entrypoint.ARGO_PARAM_SUFFIX
+        ))
+        program_args.append(_structures.InputValuePlaceholder(
+            input_name=component_input.name))
+      else:
+        raise TypeError(
+            'Only Input/OutputArtifact and parameter annotations '
+            'are supported in V2 components. '
+            'Got %s' % component_input._passing_style)
+
+    for component_output in component_spec.outputs or []:
+      if component_output._passing_style == components.OutputArtifact:
+        # For each output artifact, there'll be possibly 3 arguments passed to
+        # the user program:
+        # 1. {name of the artifact}_output_path: The actual path, or uri, of the
+        #    input artifact.
```

Comment:
nit: the comment seems to be incomplete for the 3 possibilities?

Reply:
Ah good catch! I think after the latest discussion there'll be only one argument per output artifact, then. 

Updated the comment.